### DISCUSSION TITLE
429 and Auth Errors Only on Storage Requests, But Works Fine For Database Requests

### DISCUSSION THREAD

**Discussion by johnnyicon** [supabase/supabase#30113] [Questions]
It looks like Supabase is having a challenge connecting to the Storage service.

I first noticed this issue when viewing my web app, and the images stored in Supabase were not showing.

Looking at the Network Requests, requests to Supabase for images were returning 429s. However, all other requests to the Database service were working fine. 

![CleanShot_2024-10-20_at_12](https://github.com/user-attachments/assets/3e2b46d4-de69-49c3-9768-7d64cdecaf3c)

It was strange because:
 * It only affected image requests (e.g., images being rendered via <img>
 * There are less than 10 requests being made, and my plan allows for much more (30 direct connections, 200 pooler connections)

I checked the Supabase Admin Dashboard to see if there's an issue with the Storage service. Trying to access the buckets directly, I saw this on the dashboard:

![CleanShot_2024-10-20_at_13](https://github.com/user-attachments/assets/998ccc95-23c1-4270-9bc2-a5be5f932880)

Checking the status of the Storage service, it reported Unable to connect.

![CleanShot_2024-10-20_at_123](https://github.com/user-attachments/assets/e44658eb-47ee-4f18-893b-971dee846cbc)

Taking a look at the logs, I found this error:

```
 
"error"
: [
      {
        
"message"
: 
"Authentication error, reason: \"Authentication query failed: %Postgrex.Error{message: nil, postgres: %{code: :undefined_function, file: \\\"parse_func.c\\\", hint: \\\"No function matches the given name and argument types. You might need to add explicit type casts.\\\", line: \\\"629\\\", message: \\\"function pgbouncer.get_auth(unknown) does not exist\\\", pg_code: \\\"42883\\\", position: \\\"15\\\", routine: \\\"ParseFuncOrColumn\\\", severity: \\\"ERROR\\\", unknown: \\\"ERROR\\\"}, connection_id: 1575, query: \\\"SELECT * FROM pgbouncer.get_auth($1)\\\"}\""
,
        
"name"
: 
"error"
,
        
"raw"
: 
"{\"length\":553,\"name\":\"error\",\"severity\":\"FATAL\",\"code\":\"XX000\"}"
,
        
"stack"
: 
"error: Authentication error, reason: \"Authentication query failed: %Postgrex.Error{message: nil, postgres: %{code: :undefined_function, file: \\\"parse_func.c\\\", hint: \\\"No function matches the given name and argument types. You might need to add explicit type casts.\\\", line: \\\"629\\\", message: \\\"function pgbouncer.get_auth(unknown) does not exist\\\", pg_code: \\\"42883\\\", position: \\\"15\\\", routine: \\\"ParseFuncOrColumn\\\", severity: \\\"ERROR\\\", unknown: \\\"ERROR\\\"}, connection_id: 1575, query: \\\"SELECT * FROM pgbouncer.get_auth($1)\\\"}\"\n    at Parser.parseErrorMessage (/app/node_modules/pg-protocol/dist/parser.js:283:98)\n    at Parser.handlePacket (/app/node_modules/pg-protocol/dist/parser.js:122:29)\n    at Parser.parse (/app/node_modules/pg-protocol/dist/parser.js:35:38)\n    at TLSSocket.<anonymous> (/app/node_modules/pg-protocol/dist/index.js:11:42)\n    at TLSSocket.emit (node:events:519:28)\n    at addChunk (node:internal/streams/readable:559:12)\n    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)\n    at Readable.push (node:internal/streams/readable:390:5)\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:191:23)\n    at TLSWrap.callbackTrampoline (node:internal/async_hooks:130:17)"

      }
```

Restarting the server didn't resolve the issue either.

**Comment by GaryAustin1** [comment]
As I suggested earlier in Discord, if reset did not work, please contact support.  Likely there is something wrong with your instance and users and non infrastructure employees can't help.

**Comment by dreamid27** [comment]
We got same issue @johnnyicon , any update ?
